src/elements/types/JSXElement.js

```diff
@@ -0,0 +1,41 @@
+import Expression from '../Expression';
+
+export default class JSXElement extends Expression {
+    constructor(childNodes) {
+        super('JSXElement', childNodes);
+    }
+
+    _acceptChildren(children) {
+        let closingElement = null;
+        let childrenProperty = [];
+
+        let openingElement = children.passNode('JSXOpeningElement');
+        if (!openingElement.selfClosing) {
+            children.skipNonCode();
+            while (!children.isNode('JSXClosingElement')) {
+                childrenProperty.push(children.passOneOfNode(['Literal', 'JSXExpressionContainer', 'JSXElement']));
```

Comment:
https://github.com/facebook/jsx/blob/master/AST.md#jsx-element - instead of JSXText I guess?